---
identifier: "CVE-2025-61622"
identifiers:
- "CVE-2025-61622"
- "GHSA-538v-3wq9-4h3r"
package_slug: "pypi/pyfury"
title: "Apache Pyfory python is vulnerable to deserialization of untrusted data"
description: "Deserialization of untrusted data in python in pyfory versions 0.12.0
  through 0.12.2, or the legacy pyfury versions from 0.1.0 through 0.10.3: allows
  arbitrary code execution. An application is vulnerable if it reads pyfory serialized
  data from untrusted sources. An attacker can craft a data stream that selects pickle-fallback
  serializer during deserialization, leading to the execution of `pickle.loads`, which
  is vulnerable to remote code execution.\n\nUsers are recommended to upgrade to pyfory
  version 0.12.3 or later, which has removed pickle fallback serializer and thus fixes
  this issue."
date: "2025-10-01"
pubdate: "2025-10-01"
affected_range: ">=0.1.0,<=0.10.3"
fixed_versions: []
affected_versions: "All versions starting from 0.1.0 up to 0.10.3"
not_impacted: ""
solution: "Unfortunately, there is no solution available yet."
urls:
- "https://nvd.nist.gov/vuln/detail/CVE-2025-61622"
- "https://github.com/advisories/GHSA-538v-3wq9-4h3r"
- "https://github.com/apache/fory/pull/2629"
- "https://github.com/apache/fory/commit/379b948ecae5c3b849e5bdb3997978c9a163e40b"
- "https://github.com/apache/fory"
- "https://github.com/apache/fory/releases/tag/v0.12.3"
- "https://lists.apache.org/thread/vfn9hp9qt06db5yo1gmj3l114o3o2csd"
cvss_v3: "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H"
uuid: "c31c3eea-6c10-41e0-9574-89735ba96d5a"
cwe_ids:
- "CWE-502"
- "CWE-937"
- "CWE-1035"
